model/DeliveryFactory.php

```diff
@@ -230,4 +229,69 @@ protected function createDeliveryResource(
 
         return $compilationInstance;
     }
+
+    private function createNamespacedDeliveryResource(
+        KernelClass $deliveryClass,
+        array $additionalParameters
+    ): KernelResource {
+        $deliveryId = trim($additionalParameters[DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID] ?? '');
+
+        if (!$deliveryId) {
+            throw new RuntimeException(
+                sprintf('%s must not be empty.', DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID)
+            );
+        }
+
+        $this->getDeliveryDeleteService()
+            ->execute(
+                $this->createDeliverDeleteRequest($deliveryId)
+            );
+
+        $delivery = $deliveryClass->getResource($this->createNamespacedDeliveryId($deliveryId));
+
+        $delivery->setPropertiesValues([OntologyRdf::RDF_TYPE => $deliveryClass]);
+
+        $this->getEventManager()->trigger(new ResourceCreated($delivery));
+
+        return $delivery;
+    }
+
+    private function hasNamespace(): bool
+    {
+        return $this->hasOption(static::OPTION_NAMESPACE);
+    }
+
+    private function getNamespace(): string
+    {
+        $namespace = rtrim($this->getOption(static::OPTION_NAMESPACE, ''), '#');
+
+        if ($namespace === LOCAL_NAMESPACE) {
```

Comment:
could this affect Udir as they have a specific uriProvider config?

Reply:
I'll double check.
AFAIK LOCAL_NAMESPACE is something that MUST be defined for TAO to work.